docs/_ext/validating_code_block.py

```diff
@@ -22,28 +25,58 @@ class ValidatingCodeBlock(CodeBlock):
         'type-name': directives.unchanged,
     }
     option_spec.update(CodeBlock.option_spec)
-    skip_validation = (os.getenv('SPHINX_SKIP_CONFIG_VALIDATION') or 'false').lower() == 'true'
+
+    @cached_property
+    def configs(self) -> dict:
+        _configs = dict(
+            descriptor_path="",
+            skip_validation=False,
+            validator_path="bazel-bin/tools/config_validation/validate_fragment")
+        if os.environ.get("ENVOY_DOCS_BUILD_CONFIG"):
+            with open(os.environ["ENVOY_DOCS_BUILD_CONFIG"]) as f:
+                _configs.update(yaml.safe_load(f.read()))
+        return _configs
+
+    @property
+    def descriptor_path(self) -> str:
+        return self.configs["descriptor_path"]
+
+    @property
+    def skip_validation(self) -> bool:
+        return bool(self.configs["skip_validation"])
+
+    @property
+    def validator_args(self) -> tuple:
+        args = (
+            self.options.get('type-name'),
+            '-s',
+            '\n'.join(self.content),
+        )
+        return (
+            args + ("--descriptor_path", self.descriptor_path) if self.descriptor_path else args)
```

Comment:
I'm not sure I understand how this supposed to work -- this potentially comes from a global config file, but is fragment specific? It also appears that the config file is being overwritten in `docs/build.sh`, without a way to define the descriptor. Do we need this parameter at all?

Reply:
yep - in #15229 the sphinx job is moved to a bazel `build` job

in this case the path to the descriptor file is different to the one set in `config_validation` (which can now accept this parameter) and so bazel passes the `location` via the config file here

Reply:
To see if I got it: 
 * the descriptor_path isn't fragment-specific, but if differs from the hard-coded one can be overriden
  * in this case the descriptor path needs to be overriden
  
  I don't see where in the config file the descriptor_path is being defined however, can you point me to it pls?

Reply:
in this pr the job has not yet been moved to bazel build so it isnt

here it is tho https://github.com/envoyproxy/envoy/pull/15229/files#diff-92ae908436b530a281eddfb50321c5a055ad4fa90e0194f0651b58ac39341800R206 and this is then written to the config file here https://github.com/envoyproxy/envoy/pull/15229/files#diff-b7bc9b4ee0614d05a36dbf7391fe7c20bc20f0f0a3dc67b86eb859139a333aa9R50  and consumed from there